src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.cs

```diff
@@ -644,9 +645,25 @@ public ResultWrapper<IEnumerable<FilterLog>> eth_getLogs(Filter filter)
         {
             LogFilter logFilter = _blockchainBridge.GetFilter(filter.FromBlock, filter.ToBlock,
                 filter.Address, filter.Topics);
+
             IEnumerable<FilterLog> filterLogs = _blockchainBridge.GetLogs(logFilter, fromBlock, toBlock, cancellationToken);
 
-            return ResultWrapper<IEnumerable<FilterLog>>.Success(GetLogs(filterLogs, cancellationTokenSource));
+            ArrayPoolList<FilterLog> logs = new(_rpcConfig.MaxLogsPerResponse ?? 0);
+
+            using (cancellationTokenSource)
+            {
+                foreach (FilterLog log in filterLogs)
+                {
+                    logs.Add(log);
+                }
+            }
+
+            if (logs.Count > _rpcConfig.MaxLogsPerResponse)
```

Comment:
this needs to be checked in above loop. If not you would be potentially adding milions of items to that collection.

Reply:
I'm doing same as reth has done: https://github.com/paradigmxyz/reth/blob/89e55c4830a90092bd3a3fb7f9155dfea9d2749b/crates/rpc/rpc/src/eth/filter.rs#L502

Reply:
added the changes!